### PR TITLE
AG-10326 Set default bullet colorRange based on user-defined series fill

### PR DESCRIPTION
**Pull request by olegat**
https://ag-grid.atlassian.net/browse/AG-10326

### CODE REVIEW

File: packages/ag-charts-enterprise/src/series/bullet/bulletSeriesProperties.ts [hunk 1]
```diff
@@ -103,4 +102,7 @@ export class BulletSeriesProperties extends AbstractBarSeriesProperties<AgBullet
 
     @Validate(OBJECT)
     readonly tooltip = new SeriesTooltip<AgBulletSeriesTooltipRendererParams>();
+
+    @Validate(COLOR_STRING) // Internal: Set by paletteFactory.
+    backgroundFill: CssColor = 'white';
```
Comment (olegat) [inline packages/ag-charts-enterprise/src/series/bullet/bulletSeriesProperties.ts:108]: This isn't a good solution IMHO. Is there an easy to retrieve the `themeTemplatePalette` from the module's palette factory?

Comment (alantreadway) [reply packages/ag-charts-enterprise/src/series/bullet/bulletSeriesProperties.ts:108]: The WaterfallSeries has some special handling in this respect, does this look relevant?
https://github.com/ag-grid/ag-charts/blob/65c5647df895858398b131ee07ac30e28c7eb12b/packages/ag-charts-enterprise/src/series/waterfall/waterfallModule.ts#L19-L46

Comment (olegat) [reply packages/ag-charts-enterprise/src/series/bullet/bulletSeriesProperties.ts:108]: Doesn't look relevant. The `positive`, `negative` and `total` options are documented in the API: https://charts.ag-grid.com/options/series/waterfall/#reference-AgWaterfallSeriesOptions-item

The issue that I'm facing here is that `backgroundFill` is not documented and is not meant to be. It's only used for internal implementation in _bulletSeries.ts_. The user is not meant to be able to reconfigured `backgroundFill`, but they still can simply by set `options.series[].backgroundFill`

Comment (alantreadway) [reply packages/ag-charts-enterprise/src/series/bullet/bulletSeriesProperties.ts:108]: Discussed offline, I've got the wrong end of the stick

File: packages/ag-charts-community/src/util/properties.ts [hunk 2]
```diff
@@ -72,6 +73,58 @@ export class PropertiesArray<T extends BaseProperties> extends Array {
     }
 }
 
+export class OptionalPropertiesArray<T extends BaseProperties> {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/util/properties.ts]: Maybe sync with @iMoses about the use-case for this?

Comment (olegat) [reply packages/ag-charts-community/src/util/properties.ts]: Resolved. We've decided to use an empty `PropertiesArray` instance instead.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**